### PR TITLE
exp/expm1 support int32/int64/float16 forward

### PR DESCRIPTION
**Pull request by zh794390558**
### PR types

Bug fixes
### PR changes

OPs

### Description

* exp/expm1 support int32/int64/float16 forward
* branch based on [log/log10](https://github.com/PaddlePaddle/Paddle/pull/54089)

card-71965
https://github.com/PaddlePaddle/docs/pull/5929

### CODE REVIEW

**Review by lanxianghit — DISMISSED** [review]
LGTM for dtype register

File: test/legacy_test/test_activation_op.py [hunk 1]
```diff
@@ -154,6 +154,33 @@ def init_shape(self):
...
+            with static.program_guard(
+                paddle.static.Program(), paddle.static.Program()
+            ):
+                x = [[2, 3, 4], [7, 8, 9]]
+                x = paddle.to_tensor(x, dtype='float16')
+                out = paddle.exp(x)
+                if core.is_compiled_with_cuda():
+                    place = paddle.CUDAPlace(0)
+                    exe = paddle.static.Executor(place)
+                    (res,) = exe.run(fetch_list=[out])
```
Comment (zhwesky2010) [inline test/legacy_test/test_activation_op.py:169]: 这里没有expect的值吗

Comment (zh794390558) [reply test/legacy_test/test_activation_op.py:169]: 模仿其他的单侧写的。数值的check在下面的Test_Exp_Op_Int里有。

Comment (zhwesky2010) [reply test/legacy_test/test_activation_op.py:169]: 是的。其他的单测有，但这个地方没有对应值检查

**Review by zhwesky2010 — APPROVED** [review]
LGTM

**Review by jzhang533 — APPROVED** [review]
LGTM

**Review by lanxianghit — APPROVED** [review]
LGTM for dtype register

**Pull request merged**